lib/pavilion/series/test_set.py

```diff
@@ -195,6 +203,19 @@ def make(self, build_only=False, rebuild=False, local_builds_only=False,
             self.status.set(S_STATES.ERROR, msg + ': ' + str(err.args[0]))
             raise TestSetError(msg, err)
 
+        for pattern in self._permutations:
+            for cfg_index in range(len(test_configs)):
+                cfg_name = '.'.join([
+                    test_configs[cfg_index].config['suite'],
+                    test_configs[cfg_index].config['name'],
+                    test_configs[cfg_index].config['subtitle']])
+
+                if not fnmatch(cfg_name, pattern):
+                    self._permute_indexes.append(test_configs[cfg_index])
+
+        for pindexes in self._permute_indexes:
+            test_configs.remove(pindexes)
```

Comment:
Given requests for tests 'a.b.*, c.d', the resulting test list would only include the 'a.b.*' tests. 

It's a bit worse than that though.

Let's say the user requests: 'a.b.* c.d a.b.3', given that 'a.b' has permutations 1,2, and 3 and 'c.d' has no permutations

I would expect to get five tests back: 'a.b.1', 'a.b.2', 'a.b.3', 'c.d', and 'a.b.3'.

Reply:
To handle that, you really need to do the filtering on a per-request basis. 

Which leads me to a bigger issue - While you _CAN_ put this code in TestSet, I don't think it belongs there. 

It's the job of the Resolver.load() to turn a list of test names into a list of test objects (well, ProtoTest objects). Given the problem I illustrated in the previous comment, I actually think it _HAS_ to go there. By the time the test requests have run through Resolver.load, they're disassociated from the ProtoTest objects that are produced (which you need to filter).

Now, load() doesn't keep up that association, but you have everything you need in that function to do so. That's not the case out in `TestSet.make()`.